### PR TITLE
Tbl087/adding surveys endpoints

### PR DESCRIPTION
**Pull request by andrewmil**
Added 4 endpoints to aggregate the 4 sets of calls RAS-Frontstage has too make in it's surveys blueprint
- '/get-surveys-list'
- '/access-case'
- '/download-collection-instrument'
- '/upload-collection-instrument'

The RAS-Frontstage changes to work with this PR will be linked here when they're up

To test run RAS-Frontstage-api with the appropriate environment variables and test hitting the endpoints OR wait for the RAS-Frontstage changes and test the endpoints through that

### CODE REVIEW

File: frontstage_api/controllers/collection_instrument_controller.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if response.status_code != 200:
+        logger.error('Failed to download collection instrument', collection_instrument_id=collection_instrument_id)
+        raise ApiError(url, response.status_code)
+
+    logger.debug('Successfully downloaded collection instrument', collection_instrument_id=collection_instrument_id)
+    return response.content, response.headers.items()
+
+
+def upload_collection_instrument(upload_file, case_id):
+    logger.info('Attempting to upload collection instrument', case_id=case_id)
```
Comment (gemmai95) [inline frontstage_api/controllers/collection_instrument_controller.py]: Is there a reason this log statement is info but for most other functions it is set to debug?

File: frontstage_api/controllers/case_controller.py [hunk 2]
```diff
@@ -74,3 +91,57 @@ def post_case_event(case_id, party_id, category, description):
...
+        "survey": survey,
+        "collection_instrument_size": collection_instrument_size,
+        'status': status_filter
+    }
+    logger.debug('Successfully built survey data', case_id=case['id'])
+    return survey_data
+
+
+def calculate_case_status(case):
+    case_events = case.get('caseEvents')
```
Comment (gemmai95) [inline frontstage_api/controllers/case_controller.py]: Do this and the function below need logging?

File: frontstage_api/controllers/case_controller.py [hunk 3]
```diff
@@ -74,3 +91,57 @@ def post_case_event(case_id, party_id, category, description):
         logger.error('Failed to post to case service', case_id=case_id)
         raise ApiError(url, response.status_code)
     logger.debug('Successfully posted case event', case_id=case_id)
+
+
+def check_case_permissions(party_id, case_party_id, case_id=None):
+    logger.debug('Party requesting access to case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+    if party_id != case_party_id:
+        logger.warning('Party does not have permission to access case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
```
Comment (gemmai95) [inline frontstage_api/controllers/case_controller.py]: Do we need the repeat logging here and in the error handler?

File: config.py [hunk 4]
```diff
@@ -14,11 +14,13 @@ class Config(object):
     DJANGO_CLIENT_ID = os.getenv('DJANGO_CLIENT_ID')
     DJANGO_CLIENT_SECRET = os.getenv('DJANGO_CLIENT_SECRET')
     DJANGO_BASIC_AUTH = (DJANGO_CLIENT_ID, DJANGO_CLIENT_SECRET)
+    MAX_UPLOAD_LENGTH = os.getenv('MAX_UPLOAD_LENGTH',  20 * 1024 * 1024)
```
[the remark this answers is not in the document]
Comment (andrewmil) [reply config.py]: Not sur. It's an environment variable in frontstage currently and I've just been trying to replicate the behaviour while proxying through the api

File: frontstage_api/controllers/case_controller.py (hunk 3 above)
Comment (andrewmil) [reply frontstage_api/controllers/case_controller.py]: Nope, I'll remove

File: frontstage_api/controllers/case_controller.py (hunk 2 above)
Comment (andrewmil) [reply frontstage_api/controllers/case_controller.py]: Can't hurt to add debug level logging to them anyway

File: frontstage_api/controllers/collection_instrument_controller.py (hunk 1 above)
Comment (andrewmil) [reply frontstage_api/controllers/collection_instrument_controller.py]: Will change to debug

File: frontstage_api/common/request_handler.py [hunk 5]
```diff
@@ -10,12 +10,12 @@
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def request_handler(method, url, auth=None, headers=None, json=None, data=None, fail=True):
+def request_handler(method, url, auth=None, headers=None, json=None, data=None, files=None, fail=True):
```
Comment (benjefferies) [inline frontstage_api/common/request_handler.py]: I still think the value of having this wrapper doesn't warrant the overhead I think it's adding.

We are adding more and more arguments to support different use cases thus it's looking more like [requests](http://docs.python-requests.org/en/master/) api and less of an abstraction. Where we are passing in the `method` parameter we could just be defining the method name. We could also catch `RequestException` in the `error_handling` of flask and then we would be writing roughly the same amount of code.

Comment (andrewmil) [reply frontstage_api/common/request_handler.py]: I agree, this has kind of been left over from an old way of organising things in the api that we aren't using anymore. It's used across the api though so probably best to deal with after this PR? We're making a list of the improvements we want to make across the repo once we have the functionality in

File: frontstage_api/resources/surveys/upload_collection_instrument.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+class UploadCollectionInstrument(Resource):
+
+    @staticmethod
+    @auth.login_required
+    def post():
+        case_id = request.args['case_id']
+        party_id = request.args['party_id']
+        logger.info('Attempting to upload collection instrument', case_id=case_id, party_id=party_id)
+
+        if request.content_length > app.config['MAX_UPLOAD_LENGTH']:
```
Comment (tej99) [inline frontstage_api/resources/surveys/upload_collection_instrument.py]: Might be better to have this check done in the front end?

Comment (andrewmil) [reply frontstage_api/resources/surveys/upload_collection_instrument.py]: Yeah good point I'll add that, should we have it in both maybe?

Comment (tej99) [reply frontstage_api/resources/surveys/upload_collection_instrument.py]: Don't think there's a need to have them in both

**Pull request merged**